### ISSUE TITLE
Using ES6 skate components

### ISSUE THREAD

**Issue by kevinpaxton-ct** [skatejs/skatejs#1167]
I'm probably missing something here, but I'm having a problem trying to write skate components which will run in ES6/ES2015 in modern browsers (e.g., Chrome 58).

Many modern browsers now fully support both the ES6/ES2015 language specification and natively support Custom Elements v1 and Shadow DOM v1.

Further to that, the Web Components v1 spec requires web components to be defined using the ES6/ES2015 `class` syntax:

```
class MyComponent extends skate.Component
```

This is consistent with the example here: https://skatejs.gitbooks.io/skatejs/content/docs/examples/#counter

However, when I try to use a skate component delivered to the browser in the ES6/ES2015 syntax, I get the following error:

```
component.js:1 Uncaught TypeError: Failed to construct 'HTMLElement': Please use the 'new' operator, this DOM object constructor cannot be called as a function
```

This is not the case when porting the same component to a vanilla web component which extends `HTMLElement` - that works fine.

See the following gist for quick example of how I can reproduce this issue: https://gist.github.com/kevinpaxton-ct/19f4d09f3fd75ed35a7c190a16e56b57

Any guidance appreciated. We are trying to create an environment where supporting browsers will get the ES6/ES2015 versions of web components so that we do not need any shims or es5-adapters to "backfill" support for non-class custom element definitions.

**Comment by Hotell** [comment]
Right this is not supported in skate 4.x, you should use 5.x where we ship non transpiled version of base `Component`. You can also use native browser es2015 modules without any bundler.

check it out

**Comment by treshugart** [comment]
We don't do that yet but will in https://github.com/skatejs/skatejs/issues/1142 where the `es` folder will contain code for the latest stable spec.

**Comment by kevinpaxton-ct** [comment]
We've been hoping to avoid using any shims (the `custom-elements-es5-adapter.js` shim plays havoc with Safari 10.x and including `native-shim.js` prevents us from using any ES2015 components).

Our intent was to deliver only ES2015 to the browsers which natively support Web Components v1, only delivering ES5 and polyfills to those which do not.

We have now actually managed to create an ES2015 webpack bundle of skatejs and with some massaging and trial-and-error have managed to get it working with pure ES2015 skatejs components.

**Comment by treshugart** [comment]
@kevinpaxton-ct awesome. If you have any insight on #1142, feel free.

**Comment by treshugart** [comment]
Are we able to close this now since we have that issue and you've worked around it?

**Comment by kevinpaxton-ct** [comment]
I'll mark this issue as closed as it seems we have been able to achieve what we need to by creating our own ES2015 version of the skatejs bundle (an ES2015 equivalent of https://unpkg.com/skatejs@4.6.7/dist/index-with-deps.min.js I think). This exposes a `skate` property on the `window` object.

I'll attach the file we have used here in case you want to include it in the 4.x distributions or review it, but it can be quite easily generated by creating an entry point file like this:

```
import * as skate from "skatejs/src/index";
window.skate = skate;
```

...and then running it through webpack. Note that you'll need the babel plugin "babel-plugin-transform-class-properties" in order to correctly handle the `static is = ''` code line in `component.js`.

The `skate.es6.js` file is in this zip, if this helps anybody else having this issue: [skate.es6.js.zip](https://github.com/skatejs/skatejs/files/1003394/skate.es6.js.zip)
